forge-gui/res/cardsfolder/upcoming/champions_of_tyr.txt

```diff
@@ -0,0 +1,17 @@
+Name:Champions of Tyr
+ManaCost:2 W W
+Types:Creature Angel Knight
+PT:4/3
+K:Flying
+K:Double Team
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ DBDelayedTrigger | TriggerDescription$ When CARDNAME enters the battlefield, you get a boon with "When you cast your next creature spell, that creature enters the battlefield with your choice of a +1/+1 counter, a flying counter, or a lifelink counter on it."
+SVar:DBDelayedTrigger:DB$ DelayedTrigger | Execute$ TrigAddAPI | Mode$ SpellCast | ValidCard$ Creature | ValidActivatingPlayer$ You | TriggerDescription$ When you cast your next creature spell, that creature enters the battlefield with an additional +1/+1 counter, trample counter, and vigilance counter on it.
+SVar:TrigAddAPI:DB$ Effect | RememberObjects$ TriggeredCard | ForgetOnMoved$ Stack | ReplacementEffects$ ReplaceEnter
+SVar:ReplaceEnter:Event$ Moved | ValidCard$ Card.IsRemembered | Destination$ Battlefield | ReplaceWith$ AddExtraCounter | ReplacementResult$ Updated | Description$ When you cast your next creature spell, that creature enters the battlefield with your choice of a +1/+1 counter, a flying counter, or a lifelink counter on it.
+SVar:AddExtraCounter:DB$ GenericChoice | Defined$ You | Choices$ P1P1,Flying,Lifelink
+SVar:P1P1:DB$ PutCounter | ETB$ True | Defined$ ReplacedCard |CounterType$ P1P1 | CounterNum$ 1 | SpellDescription$ +1/+1 counter
+SVar:Flying:DB$ PutCounter | ETB$ True | Defined$ ReplacedCard | CounterType$ Flying | CounterNum$ 1 | SpellDescription$ Flying counter
+SVar:Lifelink:DB$ PutCounter | ETB$ True | Defined$ ReplacedCard |CounterType$ Lifelink | CounterNum$ 1 | SpellDescription$ Lifelink counter
```

Comment:
@Northmoc i think there is a way to do that as one Effect right?

Reply:
@Simisays this line should handle the choice already:

`DB$ PutCounter | ETB$ True | Defined$ ReplacedCard | CounterType$ P1P1,Flying,Lifelink | CounterNum$ 1`

Reply:
That puts all 3 counters on the creature. You need to choose

Reply:
thats how it was implemented for https://github.com/Card-Forge/forge/blob/master/forge-gui/res/cardsfolder/f/ferocious_tigorilla.txt#L6

Reply:
Oh you're right, I forgot | Defined$ ReplacedCard | that's why I thought it didn't work that way. Thanks